src/decode-teredo.c

```diff
@@ -38,10 +38,63 @@
 #include "decode-teredo.h"
 #include "util-debug.h"
 #include "conf.h"
+#include "detect-engine-port.h"
 
-#define TEREDO_ORIG_INDICATION_LENGTH    8
+#define TEREDO_ORIG_INDICATION_LENGTH   8
+#define TEREDO_DEFAULT_PORT             3544
 
 static bool g_teredo_enabled = true;
+static int g_teredo_ports[4] = { TEREDO_DEFAULT_PORT, -1, -1, -1 };
+static int g_teredo_ports_idx = 0;
```

Comment:
Suggest naming `g_teredo_ports_cnt` instead since it represents the port count, not an index?

Shouldn't the default value be `1` instead?

Reply:
No, but it's not correct either. The default logic was taken from vxlan, but as it works differently here it doesn't make sense.

The default (no config) behaviour will be that 'any' port is used.